SwiftWisdom/Core/StandardLibrary/Numbers/Int+Extensions.swift

```diff
@@ -24,3 +24,41 @@ extension IntegerType {
         (0..<self).forEach { _ in closure() }
     }
 }
+
+extension IntegerType {
+    public func ip_toMagnitudeString(decimalPlaces decimals: Double = 1) -> String {
+        
+        print(self)
+        
+        guard self > 999 else { return "\(self)" }
+        
+        let likes = Double(self.toIntMax())
+        
+        let units = ["K", "M", "B", "T"]
+        var magnitude: Int = Int(log10(likes) / 3.0) // the order of magnitude of our like count in thousands
+        
+        // this formula rounds the likes to the nearest 1000s magnitude with X decimal points
+        // E.G. - 75633 becomes 75.63
+        // if X is 2
+        // pow(1000.0, Double(magnitude)) = 1000^1 = 1000
+        // 10 ^ 2 * likes / 1000 = 7563.3
+        // round(7563.3) = 7563
+        // 7563 / 10 ^ 2 = 75.63
+        var roundedLikes = round(pow(10.0, decimals) * likes / pow(1000.0, Double(magnitude))) / pow(10.0, decimals)
+        
+        // if rounding brings our display number over 1000, divide by 1000 and then bump the magnitude
+        if roundedLikes >= 1000 {
```

Comment:
Is there a chance it will be over 2000 units? Would something like this be more applicable:

```Swift
while roundedLikes >= 1000 {
    roundedLikes /= 1000
    magnitude += 1
}
```